cmd/debos/debos.go

```diff
@@ -243,7 +243,34 @@ func main() {
 		return
 	}
 
+	if !runInFakeMachine && !fakemachine.InMachine() {
```

Comment:
looks like a spurious change ? why the additional check of `!runInFakeMachine` ?

Reply:
This condition governs PreNoMachine(), which should only be called when we don't intend to run in the fakemachine and we aren't running in the fakemachine. If I removed !runInFakeMachine, PreNoMachine would run on the host before PreMachine when we intend to run in a fakemachine. The condition used to be implicit because the block was after the bailout point for the runInFakeMachine case (that block returns); I had to move it forward so that I could give a solid consistent point at which CheckEnv runs (i.e. _after_ PreMachine _and_ PreNoMachine).

Reply:
Yeah this seems correct.. The logic on what runs at a give point has always been a bit... wobbly.. but i don't think you made it worse ;) something to refactor in the future though